storage/misc.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package storage
+
+import (
+	"strings"
+	"time"
+)
+
+// numIntervals returns the number of intervals between [start, end] for a given
+// windowSize.
+func numIntervals(startInclusive, endInclusive time.Time, window time.Duration) int {
+	if window == 0 || endInclusive.Before(startInclusive) {
+		return 0
+	}
+
+	return 1 + int((endInclusive.Sub(startInclusive))/window)
+}
+
+func sanitizeIdentifierForMetrics(input string) string {
+	return strings.Replace(input, ":", "_", -1)
```

Comment:
Probably best to sanitize everything if we're going as far to run it through a function, maybe something like:
```go
func sanitizeMetricsTagValue(value string) string {
  buf := bytes.NewBuffer(make([]byte, 0, len(value)))
  validRanges := [][2]int{{int('a'), int('z')}, {int('A'), int('Z')}, {int('0'), int('9')}}
  validLast := false
  for _, ch := range value {
    validCurr := false
    for i := 0; !validCurr && i < len(validRanges); i++ {
      for j := validRanges[i][0]; j < validRanges[i][1]; j++ {
        if rune(j) == ch {
          validCurr = true
          break
        }
      }
    }
    if validCurr {
      buf.WriteRune(ch)
    } else if validLast {
      buf.WriteRune('_')
    }
    validLast = validCurr
  }
  if l := buf.Len(); l > 0 && !validLast {
    buf.Truncate(l-1)
  }
  return buf.String()
}
```